sdks/java/io/mqtt/src/main/java/org/apache/beam/sdk/io/mqtt/MqttIO.java

```diff
@@ -502,6 +535,121 @@ public UnboundedMqttSource getCurrentSource() {
     }
   }
 
+  @AutoValue
+  public abstract static class DynamicWrite<InputT> extends PTransform<PCollection<InputT>, PDone> {
+    abstract @Nullable ConnectionConfiguration connectionConfiguration();
+
+    abstract @Nullable SerializableFunction<InputT, String> topicFn();
```

Comment:
at pipeline expansion time, one can set topicFn to always return the topic, and payloadFn to identify function for normal writes